src/tx/builder/schema.ts

```diff
@@ -317,477 +359,578 @@ export const TX_FEE_OTHER_GAS = (txType, txSize, { relativeTtl, innerTxSize }) =
   }
 }
 
+// based on https://stackoverflow.com/a/50375286/6176994
+type UnionToIntersection<Union> =
+  (Union extends any ? (k: Union) => void : never) extends ((k: infer Intersection) => void)
+    ? Intersection : never
+
+type TxElem = readonly [string, string | Field]
+| readonly [string, string | Field, string]
+| readonly [string, string | Field, string[]]
+
+type BuildTxArgBySchemaType<SchemaType> =
+  SchemaType extends typeof Field
+    ? Parameters<SchemaType['serialize']>[0]
+    : SchemaType extends keyof BuildFieldTypes
+      ? BuildFieldTypes[SchemaType]
+      : never
+
+type BuildTxArgBySchema<SchemaLine> =
+  UnionToIntersection<
+  SchemaLine extends ReadonlyArray<infer Elem>
+    ? Elem extends TxElem ? {
+      [k in Elem[0]]: BuildTxArgBySchemaType<Elem[1]> } : never
+    : never
+  >
+
+export type RawTxObject<Tx extends TxParams> = {
+  [k in keyof Tx]: Tx[k] extends number | BigNumber | string ? string: Tx[k]
+}
+
 const BASE_TX = [
-  TX_FIELD('tag', FIELD_TYPES.int),
-  TX_FIELD('VSN', FIELD_TYPES.int)
-]
+  ['tag', FIELD_TYPES.int],
+  ['VSN', FIELD_TYPES.int]
+] as const
 
 export const CONTRACT_BYTE_CODE_LIMA = [
   ...BASE_TX,
-  TX_FIELD('sourceCodeHash', FIELD_TYPES.rawBinary),
-  TX_FIELD('typeInfo', FIELD_TYPES.sophiaCodeTypeInfo),
-  TX_FIELD('byteCode', FIELD_TYPES.rawBinary),
-  TX_FIELD('compilerVersion', FIELD_TYPES.string),
-  TX_FIELD('payable', FIELD_TYPES.bool)
-]
+  ['sourceCodeHash', FIELD_TYPES.rawBinary],
+  ['typeInfo', FIELD_TYPES.sophiaCodeTypeInfo],
+  ['byteCode', FIELD_TYPES.rawBinary],
+  ['compilerVersion', FIELD_TYPES.string],
+  ['payable', FIELD_TYPES.bool]
+] as const
+export interface TxContractByteCodeLima
+  extends BuildTxArgBySchema<typeof CONTRACT_BYTE_CODE_LIMA> {}
```

Comment:
Is it actually used externally? Should be a way to don't duplicate these exports as well